joomla/joomla-cms
# [com_plugins] Do not allow to enable/disable a protected plugin

### Summary of Changes

This makes com_plugins respect the protected status of plugins, ie, when the plugin extension is protected it will not be able to be enabled/disabled.

### Testing Instructions

0. Code review
1. Apply patch
2. Check in extensions manage which plugins are protected.
3. Go to plugins and check those plugins are now protected (cannot be enabled or disabled) in the list view and also on edit plugin form.
4. Try to disable/enable protected plugins. You should NOT be able to do it.
5. Try to disable/enable unprotected plugins. You should be able to do it.

### Documentation Changes Required

None.

### B/C considerations

This could have some B/C considerations, since current protected plugins will not be able to be enabled/disabled.
Also to not have plugins etternaly disabled this will enable all protected plugins on update. This will make user disabled protected plugins enabled, which, dependening on which plugins, can have consequences.
Mantainers please check if this is acceptable or not, and in that case only for 4.0.

@mbabker i think this was what you're talking about, please check if ok.
